src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -27,6 +27,22 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
     }
   }
 
+  private def checkInsertPermission[T](db: DataAccess, agoraEntity: AgoraEntity, username: String, snapshots: Seq[AgoraEntity])(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    // if previous snapshots exist, check ownership on them; if not, check permission on the namespace
+    if (snapshots.isEmpty)
+      checkNamespacePermission(db, agoraEntity, username, AgoraPermissions(Create))(op)
+    else {
+      val ownerPerm = AgoraPermissions(Manage)
```

Comment:
fyi, Brad asked specifically to use `Create`, not `Manage`, because it gets shown in the error message.